packages/esm-appointments-app/src/appointments/scheduled/scheduled-appointments.component.tsx

```diff
@@ -1,124 +1,120 @@
 
   return (
     <>
-      {isToday && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setScheduleType(name)}>
-          <Switch name={'Scheduled'} text={t('scheduled', 'Scheduled')} />
-          <Switch name={'Honoured'} text={t('honored', 'Honored')} />
-          <Switch name={'Pending'} text={t('notArrived', 'Not arrived')} />
-          <Switch name={'CameEarly'} text={t('cameEarly', 'Came early')} />
+      {dateType === 'today' && (
+        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
+          {scheduledAppointmentPanels.map((panel) => {
+            return panel.meta.showForToday ? (
+              <Switch name={panel.meta.title} text={t(panel.meta.title, panel.meta.title)} />
+            ) : (
+              <></>
+            );
+          })}
         </ContentSwitcher>
       )}
-      {isDateInPast && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setScheduleType(name)}>
-          <Switch name={'Scheduled'} text={t('scheduled', 'Scheduled')} />
-          <Switch name={'Honoured'} text={t('honored', 'Honored')} />
-          <Switch name={'Pending'} text={t('missed', 'Missed')} />
+
+      {dateType === 'pastDate' && (
+        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
+          {scheduledAppointmentPanels.map((panel) => {
+            return panel.meta.showForPastDate ? (
+              <Switch name={panel.meta.title} text={t(panel.meta.title, panel.meta.title)} />
+            ) : (
+              <></>
+            );
+          })}
         </ContentSwitcher>
       )}
-      {isDateInFuture && (
-        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setScheduleType(name)}>
-          <Switch name={'Scheduled'} text={t('scheduled', 'Scheduled')} />
-          <Switch name={'CameEarly'} text={t('cameEarly', 'Came early')} />
+
+      {dateType === 'futureDate' && (
+        <ContentSwitcher className={styles.switcher} size="sm" onChange={({ name }) => setCurrentTab(name)}>
+          {scheduledAppointmentPanels.map((panel) => {
+            return panel.meta.showForFutureDate ? (
+              <Switch name={panel.meta.title} text={t(panel.meta.title, panel.meta.title)} />
+            ) : (
+              <></>
+            );
+          })}
         </ContentSwitcher>
       )}
-      <div className={styles.container}>
-        <AppointmentsTable {...currentConfig} />
-      </div>
+
+      <ExtensionSlot name={scheduledAppointmentsPanelsSlot}>
+        {(extension) =>
+          extension.meta.title === currentTab && (
+            <div className={styles.container}>
+              <Extension
+                state={{
+                  date,
+                  appointmentServiceType,
+                  status: extension.meta.status,
+                  statusText: extension.meta.title,
```

Comment:
I'm a little confused about why we have two of these? Also `meta.title` looks like it's always written in lower case?

Reply:
I thought I had a reason for this, but it is a little ugly, will double-check and see if I can refactor it out.

Reply:
Okay so status is the actual status to search on, and statusText is the display text.  "statusText" is pretty ugly name, so I changed it to "title".  It's in lower case because it's reference to message code.